src/dune_trace/dune_trace_stubs.c

```diff
@@ -1,6 +1,38 @@
+#include <caml/alloc.h>
+#include <caml/threads.h>
 #include <caml/fail.h>
 #include <caml/memory.h>
 #include <caml/mlvalues.h>
+#include <caml/bigarray.h>
+#include <caml/unixsupport.h>
+
+#if OCAML_VERSION_MAJOR < 5
+#define caml_uerror uerror
+#endif
+
+CAMLprim value dune_trace_write(value v_fd, value v_buf, value v_offset,
+                                value v_length) {
+  CAMLparam4(v_fd, v_buf, v_offset, v_length);
+#ifdef _WIN32
+  int fd = win_CRT_fd_of_filedescr(v_fd);
+#else
+  int fd = Int_val(v_fd);
+#endif
+  intnat offset = Long_val(v_offset);
+  intnat length = Long_val(v_length);
+  void *buf = Caml_ba_data_val(v_buf);
+  ssize_t written;
+
+  caml_release_runtime_system();
+  written = write(fd, (char *)buf + offset, length);
```

Comment:
Which `write` is this?

Reply:
This comes from `unistd.h` right? I couldn't work out which header this came from. If so won't we need something different for windows?

Reply:
Alternatively we can copy `mdb.c` with:
```c
#ifdef _MSC_VER
#include <io.h>
typedef SSIZE_T	ssize_t;
#else
#include <unistd.h>
#endif
```
and
```
#define write _write
```
in there